Build

LeSpocky opened the pull request:
Some more or less trivial build fixes. Makes it a little easier for others to try this out and play with it.

LeSpocky commented on mqtt-clock.c:
```diff
@@ -168,14 +168,13 @@ int main(int argc, char *argv[]) {
     int ret;
-    mosqagent_add_idle_call(agent, clock_idle);
+    ret = mosqagent_add_idle_call(agent, clock_idle);
 
```
I think this is a real bugfix and should be in an extra commit.

penguineer commented on mqtt-clock.c:
```diff
@@ -167,14 +168,13 @@ int main(int argc, char *argv[]) {
 	     mosqagent_strerror(ret));
   }
 
-  int ret;
```
This is pure clean-up, should also be an extra commit.

penguineer commented:
Thanks for the PR. I have a not-yet-pushed branch of a further development state for this library, which I will check against your proposal. 

I'll comment when this is done (to avoid unnecessary work).

penguineer reviewed (changes requested):
Please split this commit into:
1) One commit fixing the code
2) One commit changing the build environment.

Either in this PR or as a new PR. I'll integrate the result.
Thanks.

LeSpocky committed:
build: Add -Wall to CFLAGS

This is nice to spot all kinds of programming mistakes, which could lead
to strange behaviour, hard to debug.

LeSpocky committed:
clock: Remove declaration for unused variable

Exposed through new compiler warnings.

LeSpocky commented:
Splitted the requested commit. Note: there are still compiler warnings left, I did not fix, all of the type _return-type_ aka "control reaches end of non-void function".

The whitespace changes in 6166b04bca14615feb81db83c6361641f94b4433 were done automatically by my editor. Hope that doesn't disturb too much.

penguineer commented:
Thanks, merged and closed.

(nb: Most editors can be set to fix whitespace issues only in lines which have been changed anyway.)
